### ISSUE TITLE
XAML STYLER

### ISSUE THREAD

**Issue by Jay-o-Way** [microsoft/WinUI-Gallery#1422]
### Issue type

enhancement

### Which version of the app?

WinUI 3 Gallery

### Description

Just wanted to create an issue in order to have a handle to let **XAML Styler** run over a few folders.

- [ ] ControlPages
- [ ] Controls
- [ ] Navigation
- [ ] SamplePages
- [ ] TabViewPages
- [ ] WinUIGallery (*.xaml)

**Comment by Jay-o-Way** [comment]
I can do this 😇

**Comment by Jay-o-Way** [comment]
Or... Let the workflow do this.
Like https://github.com/files-community/Files/issues/13396

**Comment by Jay-o-Way** [comment]
@chingucoding any thoughts?

**Comment by karkarl** [comment]
Thanks so much for tackling this @Jay-o-Way! As you know already cleaning up the XAML is in the [backlog](https://github.com/orgs/microsoft/projects/368/views/1?pane=issue&itemId=27267745) as part of the efforts to clean up the codebase. 

I like the idea of a `Settings.XamlStyler` file should we have future customizations, and so it's easier to reference implemented rules.

**Comment by Jay-o-Way** [comment]
Thanks for the positive comment @karkarl :) Imma need a bit more specific answers to adress this, mainly the choice between a one-time (manual) execution or the workflow solution. (easy versus not-so-easy, resp.)

**Comment by Jay-o-Way** [comment]
> I think having a config for this is also necessary to avoid mixed formatting based on who is doing it.

Well, the default settings seem fine to start with. I think it's more a matter of: is the plug-in installed and being used in Visual Studio? But then, if it's done via a central workflow, this is a non-issue, because the changes are not made by people. Right? :)

> is there a way to work around that formatting of text?

You mean the unintended changes with spaces? I believe setting the `space=preserve` is the solution.

**Comment by Jay-o-Way** [comment]
@niels9001 Currently, there are a lot of RichTextBlock + Paragraph constructions. Mostly used for formatting text and including Hyperlinks. Adjusting them to add `space=preserve`, is a **huge** _p.i.t.a._ and the resulting xaml is not pretty. I think it would be a lot more efficient to use a **MarkdownTextblock**. wdyt?

**Comment by Jay-o-Way** [comment]
Also, cases like the following might be too much XAML for simple text
```xaml
<RichTextBlock
 x:Name="ErrorTextBlock"
 Grid.Row="1"
 Margin="0,0,0,12"
 Foreground="{ThemeResource SystemFillColorCriticalBrush}"
 IsTextSelectionEnabled="True"
 Visibility="Collapsed">
 <Paragraph xml:space="preserve">
 This sample requires a later version of Windows to be fully functional.
 </Paragraph>
</RichTextBlock>
```

**Comment by karkarl** [comment]
Linking this to this task in backlog #1279

**Comment by Jay-o-Way** [comment]
Example of pr: https://github.com/files-community/Files/pull/15115

**Comment by niels9001** [comment]
@Jay-o-Way Were you still planning to look at this? 100% agree we should do this, because not having this makes reviewing XAML PRs a pain :(

**Comment by Jay-o-Way** [comment]
Not really. Sometimes things seem easy at first glance, but are just insanely complex when you really dig into it. I honestly gave up a long time ago.
